launch/servable.py

```diff
@@ -0,0 +1,45 @@
+from launch.logger import logger
+
+
+def default_make_request(_, fn, args, kwargs):
+    return fn(*args, **kwargs)
+
+
+try:
+    # Keep in line with Scale Launch serverside code!
+    from hosted_model_inference.inference.service_requests import make_request
+except ImportError:
+    logger.info(
+        "Unable to import serverside request maker; falling back to local request"
+    )
+    make_request = default_make_request
+
+
+def step_decorator(fn):
+    """
+    Decorator to mark a function as a separate Servable.
+    """
+
+    # Identifiers will just be the name of the function. Must be unique. TODO enforce this uniqueness
+    fn_name = fn.__name__
+
+    def modified_fn(*args, **kwargs):
+        return make_request(fn_name, fn, args, kwargs)
+
+    return modified_fn
+
+
+def step_decorator_class(class_obj):
```

Comment:
any reason we need both the fn version and the class?

Reply:
tbh I think we could have just checked whether something was a function or a class

having said that, I think the pr is not what we're gonna end up going with (I think we're gonna have people subclass some `Servable` class), I should have closed this pr lol mb